src/plugins/vocablink.js

```diff
@@ -0,0 +1,67 @@
+let redact;
+
+const VOCABLINK_RE = /^\[v ([^\]]+)\](?:\[([^\]]+)\])?/;
+const VOCABLINK = 'vocablink';
+
+/**
+ * Plugin that adds support for Curriculum Builder's vocablinks.
+ *
+ * Note that vocab links are ONLY supported in redaction mode; rendering them
+ * out requires CurriculumBuilder database access, so they can only be rendered
+ * by Curriculum Builder itself.
+ *
+ * see https://github.com/mrjoshida/curriculumbuilder/blob/bf74aa5/curriculumBuilder/vocablinks.py
+ * @requires restorationRegistration
+ */
+module.exports = function vocablink() {
+  if (this.Parser) {
+    const Parser = this.Parser;
+    redact = Parser.prototype.options.redact;
+    Parser.prototype.inlineTokenizers[VOCABLINK] = tokenizeVocablink;
+
+    Parser.prototype.restorationMethods[VOCABLINK] = function (add, node, content) {
+      let value = `[v ${node.vocabword}]`;
+      if (content) {
+        value += `[${content}]`
+      }
+      return add({
+        type: 'rawtext',
+        value 
+      });
+    }
+
+    // Run it just before `html`
+    const methods = Parser.prototype.inlineMethods;
+    methods.splice(methods.indexOf('html'), 0, VOCABLINK);
```

Comment:
where can I see the list of methods we are splicing into?

Reply:
[here](https://github.com/remarkjs/remark/tree/master/packages/remark-parse#parserinlinemethods)! The unifiedjs documentation is in general quite good